### PR TITLE
Add PHP7.1 to test matrix.

### PR DESCRIPTION
**Pull request by markstory**
Get started on PHP7.1 support for 2.x

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ php:
   - 5.5
   - 5.6
   - 7.0
+  - 7.1
```
Comment (ravage84) [inline .travis.yml]: Could we put it into a ``allow_failures`` section for the time being?

Comment (markstory) [inline .travis.yml]: I was hoping to get the tests passing in this pull request as well unless it is a huge diff.

File: lib/Cake/Test/Case/Controller/Component/Auth/ControllerAuthorizeTest.php [hunk 2]
```diff
@@ -47,15 +47,14 @@ public function setUp() {
...
- * @throws PHPUnit_Framework_Error
  * @return void
  */
 	public function testControllerTypeError() {
 		try {
 			$this->auth->controller(new StdClass());
+			$this->fail('No exception thrown');
 		} catch (Throwable $t) {
-			throw new PHPUnit_Framework_Error($t);
+			$this->assertTrue(true, 'Exception was raised');
```
Comment (ravage84) [inline lib/Cake/Test/Case/Controller/Component/Auth/ControllerAuthorizeTest.php]: Clever 😼

Comment (markstory) [inline lib/Cake/Test/Case/Controller/Component/Auth/ControllerAuthorizeTest.php]: Not ideal, but given the PHPUnit version and PHP versions we are supporting it works.

**Pull request merged**